pkg/utils/etcdutil/etcdutil.go

```diff
@@ -689,12 +687,11 @@ func (lw *LoopWatcher) initFromEtcd(ctx context.Context) int64 {
 		watchStartRevision int64
 		err                error
 	)
-	ticker := time.NewTicker(defaultLoadFromEtcdRetryInterval)
+	ticker := time.NewTicker(defaultEtcdRetryInterval)
 	defer ticker.Stop()
-	ctx, cancel := context.WithTimeout(ctx, lw.loadTimeout)
-	defer cancel()
-
 	for i := 0; i < lw.loadRetryTimes; i++ {
+		ctx, cancel := context.WithTimeout(ctx, lw.loadTimeout)
+		defer cancel()
```

Comment:
put it at the end of loop?

Reply:
if without defer, it will need to be added in three places